enterprise/backend/src/metabase_enterprise/workspaces/merge.clj

```diff
@@ -60,23 +75,44 @@
                    {:error e}))))]
 
     (when-not error
-      (t2/delete! :model/WorkspaceTransform :ref_id ref_id))
+      (t2/delete! :model/WorkspaceTransform :ref_id ref_id)
+      (create-merge-transform-history!
+       (:op result)
+       {:workspace_merge_id         workspace-merge-id
+        :workspace_id               (:id workspace)
+        :workspace_name             (:name workspace)
+        :transform_id               (:global_id result)
+        :workspace_transform_ref_id ref_id
+        :commit_message             commit-message
+        :creator_id                 actor-id}))
 
     result))
 
 ;; TODO (crisptrutski 2025-12-XX): When there are more entity types support, this should update those too.
 (defn merge-workspace!
   "Make all the transforms in the Changeset public, i.e. create or update the relevant model/Transform entities.
-   This should also clear the entire Changeset, as it no longer has any changes."
-  [ws-id]
+   This should also clear the entire Changeset, as it no longer has any changes.
+
+   - ws-id: ID of the workspace to merge
+   - actor-id: user performing the merge
+   - commit-message: description of the merge"
+  [ws-id actor-id commit-message]
   ;; Perhaps we want to solve the N+1?
   ;; This will require reworking the lifecycle for validating and recalculating dependencies for the transforms.
 
   (t2/with-transaction [tx]
-    (let [savepoint (.setSavepoint ^Connection tx)
+    (let [workspace       (t2/select-one [:model/Workspace :id :name] :id ws-id)
```

Comment:
Ditto, rather change the function signature to take a previously queried workspace instead.